packages/peregrine/lib/mixins/AuthModal/useAuthModal.js

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useUserContext } from '../../context/user';
+
+const UNAUTHED_ONLY = ['CREATE_ACCOUNT', 'FORGOT_PASSWORD', 'SIGN_IN'];
+
+export const useAuthModal = props => {
+    const {
+        closeDrawer,
+        showCreateAccount,
+        showForgotPassword,
+        showMainMenu,
+        showMyAccount,
+        view
+    } = props;
+
+    const [username, setUsername] = useState('');
```

Comment:
This state isn't used in the hook anywhere, should it exist in the `AuthModal` component only?

Reply:
> but it's totally fine. I want the hook to do it

- Sanford, 2019